src/Symfony/Component/HttpKernel/EventListener/DumpListener.php

```diff
@@ -26,18 +27,25 @@ class DumpListener implements EventSubscriberInterface
 {
     private $cloner;
     private $dumper;
+    private $serverDumper;
 
-    public function __construct(ClonerInterface $cloner, DataDumperInterface $dumper)
+    public function __construct(ClonerInterface $cloner, DataDumperInterface $dumper, ServerDumper $serverDumper = null)
     {
         $this->cloner = $cloner;
         $this->dumper = $dumper;
+        $this->serverDumper = $serverDumper;
     }
 
     public function configure()
     {
         $cloner = $this->cloner;
         $dumper = $this->dumper;
 
+        if ($this->serverDumper) {
+            $this->serverDumper->setWrappedDumper($dumper);
```

Comment:
can't this be replaced by constructor injection?
would be better than making the server dumper mutable?
(otherwise, lets clone the service before mutating?)

Reply:
Not sure, but IIRC I made this so we're safe in case someone tweaked the injected dumper in this listener. Not sure if we really care, though. I've simplified everything to inject the server dumper if available, cli dumper otherwise (cli dumper is already the wrapped dumper).